### PR TITLE
vendor: create the vendor/ directory and include all dependencies

### PR DESCRIPTION
**Pull request by nixpanic**
The `vendor/` directory can now get created. Downloading all dependencies in advance will speed up building a little bit.

A few simple CI jobs have been added as well, these should guard that the `vendor/` contents is consistent.

### CODE REVIEW

File: .github/workflows/codespell.yml [hunk 1]
```diff
@@ -16,6 +16,6 @@ jobs:
       - name: codespell
         uses: codespell-project/actions-codespell@master
         with:
-          skip: .git
+          skip: .git,vendor
```
Comment (Rakshith-R) [inline .github/workflows/codespell.yml:19]: Can please add go.mod and go.sum too 

refer https://github.com/csi-addons/kubernetes-csi-addons/runs/4604315553?check_suite_focus=true

**Review by Rakshith-R — COMMENTED** [review]
Do we really need vendor directory ? 
I have not seen operators/controllers using vendor directory

**Comment by Rakshith-R** [comment]
> The `vendor/` directory can now get created. Downloading all dependencies in advance will speed up building a little bit.

This will need some changes in Dockerfile and ./build/Containerfile.sidecar too,  since we just copy go.mod, go.sum and download dependencies. 

https://github.com/csi-addons/kubernetes-csi-addons/blob/0daf3c935289d306346fd57ec398c1c58617eba1/Dockerfile#L6-L10
https://github.com/csi-addons/kubernetes-csi-addons/blob/0daf3c935289d306346fd57ec398c1c58617eba1/build/Containerfile.sidecar#L5-L10

**Comment by nixpanic** [comment]
> Do we really need vendor directory ? I have not seen operators/controllers using vendor directory

Yes, mostly for faster building. It helps while doing an initial clone of the repository, not all dependencies need to be downloaded.

**Comment by nixpanic** [comment]
> This will need some changes in Dockerfile and ./build/Containerfile.sidecar too, since we just copy go.mod, go.sum and download dependencies.

Ah, good point!

File: .github/workflows/codespell.yml (hunk 1 above)
Comment (nixpanic) [reply .github/workflows/codespell.yml]: ok, done!

**Comment by Rakshith-R** [comment]
add ` -mod vendor ` to go build cmd ?

**Comment by nixpanic** [comment]
> add `-mod vendor` to go build cmd ?

Yes, seems to be needed. I expected that new golang version pick it up automatically, it seems not to be the case.

**Review by yati1998 — APPROVED** [review]

File: .github/workflows/test-golang.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+name: Run Golang tests
+
+on:
+  pull_request:
+    branches:
+      - "*"
+
+jobs:
+  go_mod_verify:
```
Comment (Madhu-1) [inline .github/workflows/test-golang.yaml:10]: update mergify rules for new jobs?

Comment (nixpanic) [reply .github/workflows/test-golang.yaml:10]: yes, but only after this is merged by Mergify

**Review by Madhu-1 — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

**Review by humblec — APPROVED** [review]

**Pull request merged**